timesketch/lib/analyzers/contrib/misp_analyzer.py

```diff
@@ -0,0 +1,173 @@
+
+    def __init__(self, index_name, sketch_id, timeline_id=None, **kwargs):
+        """Initialize the Analyzer.
+
+        Args:
+            index_name: OpenSearch index name
+            sketch_id: The ID of the sketch.
+            timeline_id: The ID of the timeline.
+        """
+        super().__init__(index_name, sketch_id, timeline_id=timeline_id)
+        self.misp_url = kwargs.get("misp_url")
+        self.misp_api_key = kwargs.get("misp_api_key")
+        self.total_event_counter = 0
+        self.request_set = set()
+        self.result_dict = dict()
+
+    @staticmethod
+    def get_kwargs():
+        """Get kwargs for the analyzer.
+
+        Returns:
+            Info to connect to MISP.
+        """
+
+        misp_url = current_app.config.get("MISP_URL")
+        misp_api_key = current_app.config.get("MISP_API_KEY")
+
+        if not misp_api_key or not misp_url:
+            logger.error("MISP conf not found")
+            return []
+
+        matcher_kwargs = [{"misp_url": misp_url, "misp_api_key": misp_api_key}]
+        return matcher_kwargs
+
+    def get_misp_attributes(self, value, attr):
+        """Search event on MISP.
+
+        Args:
+            value:  Can be a valur for: sha1 - sha256 - md5 - filename.
+            attr:  type of the value.
+
+        Returns:
+            List of matching MISP attibutes.
+        """
+        results = requests.post(
+            f"{self.misp_url}/attributes/restSearch/",
+            data={"returnFormat": "json", "value": value, "type": attr},
+            headers={"Authorization": self.misp_api_key},
+            verify=False,
+        )
+
+        if results.status_code != 200:
+            msg_error = "Error with MISP query: Status code"
+            logger.error("{} {}".format(msg_error, results.status_code))
+            return []
+        result_loc = results.json()
+        if "name" in result_loc:
+            if "Authentication failed." in result_loc["name"]:
+                logger.error("Bad API key. Please change it.")
+                return []
+        if not result_loc["response"]["Attribute"]:
+            return []
+
+        self.total_event_counter += 1
+
+        return result_loc["response"]["Attribute"]
+
+    def mark_event(self, event, result, attr):
+        """Anotate an event with data from MISP result.
+
+        Add a comment to the event.
+
+        Args:
+            event:  The OpenSearch event object that contains type of value we search
+                    for and needs to be tagged or to add an attribute.
+            result:  Dictionnary with results from MISP.
+            attr:  type of the current value.
+        """
+
+        msg = "Event that match files:   "
+        for misp_attr in result:
+            info = misp_attr["Event"]["info"]
+            id_event = misp_attr["Event"]["id"]
+            msg += f'"Event info": "{info}"'
+            msg += f"- \"url\": {self.misp_url + '/events/view/' + id_event} || "
```

Comment:
```suggestion
            msg += f'- "url": {self.misp_url}/events/view/{id_event} || '
```

Reply:
Fixed in c0fd536f2fea435e66ecd4d5cf919714474952e8.